src/ngraph/pass/memory_layout.cpp

```diff
@@ -38,16 +38,48 @@ bool pass::MemoryLayout::run_on_function(shared_ptr<ngraph::Function> function)
     MemoryManager mm(m_alignment);
     for (shared_ptr<Node> node : function->get_ordered_ops())
     {
+        std::map<descriptor::Tensor*, descriptor::Tensor*> inplace_outputs;
+        std::set<descriptor::Tensor*> reused_inputs;
+
+        if (auto op = std::dynamic_pointer_cast<op::Op>(node))
+        {
+            auto op_annotations = std::dynamic_pointer_cast<op::Op>(node)->get_op_annotations();
+            if (op_annotations)
+            {
+                for (auto oi_pair : op_annotations->get_in_place_oi_pairs())
+                {
+                    auto output = &node->get_outputs().at(oi_pair.first).get_tensor();
+                    auto input = &node->get_inputs().at(oi_pair.second).get_tensor();
+
+                    if (node->liveness_free_list.count(input) != 0 &&
+                        node->liveness_new_list.count(output) != 0)
+                    {
+                        NGRAPH_DEBUG << input->get_name() << " will be reused for "
+                                     << output->get_name();
+                        inplace_outputs.insert(make_pair(output, input));
```

Comment:
You can use `.insert({output, input})` which I think looks better, but it is up to you.